File: frontend/src/modules/auth/validators/EmpresaValidator.ts
/**
 * EmpresaValidator — Chain of Responsibility node for company name presence (FE).
 * Extends BaseValidator (core). SOLID: SRP·OCP·LSP.
 */

import { BaseValidator, type ValidationResult } from '../../../core/base/BaseValidator'

export class EmpresaValidator extends BaseValidator {
  validate(data: unknown): ValidationResult {
    const empresa = String((data as { empresa?: string })?.empresa ?? '').trim()
    if (!empresa) {
      return {
        isValid: false,
        errors: ['El nombre de la empresa es obligatorio.'],
        field: 'empresa',
      }
    }
    return { isValid: true, errors: [], field: 'empresa' }
  }
}

File: backend/apps/authentication/validators/empresa_validator.py
"""
EmpresaValidator — Chain of Responsibility node validating company name presence (SRP).
Extends BaseValidator (S1). Pattern: Chain of Responsibility. SOLID: SRP·OCP·LSP.
"""

from __future__ import annotations

from core.base.base_validator import BaseValidator, ValidationResult


class EmpresaValidator(BaseValidator):
    def validate(self, data: dict) -> ValidationResult:
        empresa = (data.get("empresa") or "").strip()
        if not empresa:
            return ValidationResult(
                is_valid=False,
                errors=["El nombre de la empresa es obligatorio."],
                field_name="empresa",
            )
        return ValidationResult(is_valid=True)
